core/src/main/scala/org/bitcoins/core/util/NumberUtil.scala

```diff
@@ -188,6 +188,31 @@ trait NumberUtil extends BitcoinSLogger {
     }
 
   }
+
+  /**
+    * Implements this check for overflowing for [[org.bitcoins.core.protocol.blockchain.BlockHeader.nBits]]
+    * [[https://github.com/bitcoin/bitcoin/blob/2068f089c8b7b90eb4557d3f67ea0f0ed2059a23/src/arith_uint256.cpp#L220 bitcoin core check]]
+    * @param nBits
+    * @return
+    */
+  def isNBitsOverflow(nBits: UInt32): Boolean = {
```

Comment:
What's meant by overflow in this context? If the nbits overflow it's always invalid?

Reply:
We have an overflow check here in `arith_uint256.cpp` https://github.com/bitcoin/bitcoin/blob/fa6180188b8ab89af97860e6497716405a48bab6/src/arith_uint256.cpp#L219